spec/integration/aws_nat_gateway_spec.rb

```diff
@@ -7,13 +7,17 @@
   when_the_chef_12_server 'exists', organization: 'foo', server_scope: :context do
     with_aws 'with a VPC' do
       purge_all
-      setup_public_vpc
```

Comment:
@dheerajd-msys we can use `setup_public_vpc` def to avoid repetition. This can be achieve by extracting subnet_id as `              sub_id = test_public_subnet.aws_object.id` in `except_recipe{`